internal/engine/upper.go

```diff
@@ -135,17 +147,32 @@ func (u Upper) Up(ctx context.Context, services []model.Service, watchMounts boo
 				if !ok {
 					return nil
 				}
-				logger.Get(ctx).Info("files changed, rebuilding %v", service.Name)
 
 				var changedPaths []string
 				for _, e := range events {
 					path, err := filepath.Abs(e.Name)
 					if err != nil {
 						return err
 					}
-					changedPaths = append(changedPaths, path)
+					isIgnored, err := eventFilter.IsIgnored(path, false)
+					if err != nil {
+						return err
+					}
+					if !isIgnored {
+						changedPaths = append(changedPaths, path)
+					}
+				}
+				if len(changedPaths) > 0 {
+					var changedPathsToPrint []string
+					if len(changedPaths) > maxChangedFilesToPrint {
+						changedPathsToPrint = append(changedPaths[:maxChangedFilesToPrint], "...")
+					} else {
+						changedPathsToPrint = changedPaths
```

Comment:
can't you just declare in ln 166 as `changedPathsToPrint := changedPaths`? then you can get rid of this extra `else`

Reply:
Yep, that'd work.

It's not a big deal either way, but I generally prefer not changing variables' values / reusing variables when reasonable. This way the variable is only set once, and keeps its value. If I see code that declares a variable and then modifies it on the next line, I'll worry about whether they're changing it later in the function and when I get to the code that actually uses it, worry about whether it got mutated on the way.

Of course, even as written in this PR, it could get modified later, and the block is so short it doesn't really matter in this case.